x/qgb/keeper/keeper_data_commitment.go

```diff
@@ -16,6 +16,10 @@ func (k Keeper) GetCurrentDataCommitment(ctx sdk.Context) (types.DataCommitment,
 	beginBlock := uint64(ctx.BlockHeight()) - k.GetDataCommitmentWindowParam(ctx) + 1
 	// for a data commitment window of 400, the ranges will be: 1-400;401-800;801-1200
 	endBlock := uint64(ctx.BlockHeight())
+
+	if !k.CheckLatestAttestationNonce(sdk.UnwrapSDKContext(ctx)) {
```

Comment:
perhaps silly question, but do you know why are we unwrapping the context here, but not for all keeper methods?

Reply:
wooo, good catch. the context at that level is already unwrapped, no need to do it. 
Fixed: [46c2ae9](https://github.com/celestiaorg/celestia-app/pull/1638/commits/46c2ae953167d4423d8c3560e51530212e7eb0ae)